src/test/java/com/couchbase/connect/kafka/sink/SubDocumentWriterTest.java

```diff
@@ -220,7 +221,7 @@ public void addsToArrayUnique() {
 
     }
 
-    @Test(expected = DocumentDoesNotExistException.class)
+    @Test(expected = CannotRetryException.class)
```

Comment:
this looks fishy, since its changing the meaning of the test and also the semantics possibly

Reply:
i agree with you on this one, but i am not sure how to bubble the original error when you use the retrybuilder? The code retries a documentdoesnotexist by creating the doc and retrying the same command. ideally i would bubble the actual error if it persists.

Reply:
right well I think if we agree the change is the one we want to make we need to revisit the test case and see if its still legit and if so change it/rename it to reflect its new meaning

Reply:
i could try to change the response in the retry to not fail the second time?